### PR TITLE
Adds GDAL Vector drivers description

### LINKED ISSUE #6051: List availabale GDAL vector drivers (fix #18738) (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#38461
Author: @alexbruy
QGIS version: 3.16

**List availabale GDAL vector drivers (fix #18738)**

## Description
In QGIS settings there is a GDAL tab with list of available GDAL raster drivers and their capabilities. If several drivers can be used to open data, user can disable some drivers to make sure that data will be opened with required driver. Similar situation with vector formats. For example there are FileGDB and OpenFileGDB drivers to work with .gdb files, but if data file has version 9.x, FileGDB won't be able to open it. But there are no UI with list of available vector drivers.

This pull-requst adds "Vector Drivers" tab to the GDAL section in the QGIS options dialog. At this tab user can see all available vector drivers and their capabilities as well as uncheck drivers which should not be used to open vector data.

Raster drivers
raster

Vector drivers
vector

Fixes #18738.

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by SrNetoChan**
Fixes #6051

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 1]
```diff
@@ -728,14 +728,14 @@ GDAL Settings
 
 `GDAL <https://gdal.org>`_ is a data exchange library for vector and raster
 files. It provides drivers to read and or write data in different formats.
-The :guilabel:`GDAL` tab currently exposes the drivers for raster formats with
-their capabilities.
+The :guilabel:`GDAL` tab currently exposes the drivers for raster and vector
+formats with their capabilities.
 
 GDAL driver options
```
Comment (DelazJ) [inline docs/user_manual/introduction/qgis_configuration.rst:734]:
```suggestion
Raster driver options
```

**Review by DelazJ — COMMENTED** [review]
@SrNetoChan  you'd need to update (at least) the note in introduction at https://docs.qgis.org/testing/en/docs/user_manual/managing_data_source/opening_data.html also

PS: what is the default theme/desired env for screenshots?

**Comment by SrNetoChan** [comment]
> @SrNetoChan  you'd need to update (at least) the note in introduction at https://docs.qgis.org/testing/en/docs/user_manual/managing_data_source/opening_data.html also

Good catch. Will do.

> 
> PS: what is the default theme/desired env for screenshots?

Well, we used to have Ubuntu, but seem to remember seeing some really different screenshots in the past and a discussion saying that we no longer need to follow that rule.

I still think it would look much better to have all screenshots done in the same operaring system. Probably it would be more comfortable for the majority of our users if we use windows. On the other  hand, I have moved to kubuntu and doing screenshots in another system is too much work.

Anyway, if we strongly decide for Ubuntu or any other, I will create a VM just for it.

**Comment by DelazJ** [comment]
> Anyway, if we strongly decide for Ubuntu or any other, I will create a VM just for it.

I'm not in favor of any strong environment unless we find someone dedicated to this (never succeeded with the automated tool). When the screenshot is out of date, the contributor fixes that with what they have in hands (macOS, Windows, Linux...).

It's the commit message `Use **default theme** screenshots` that made me confused, because afaics two of the screenshots are not outdated, and they were made with Ubuntu default theme if I'm not wrong.

**Comment by SrNetoChan** [comment]
> It's the commit message `Use **default theme** screenshots` that made me confused, because afaics two of the screenshots are not outdated, and they were made with Ubuntu default theme if I'm not wrong.

I see the confusion, in the first commit I have used dark mode.

**Comment by SrNetoChan** [comment]
I have just opted to update the other two screenshots to make the all subsection look the same. But, I can revert if you prefer.

**Comment by DelazJ** [comment]
> I see the confusion, in the first commit I have used dark mode.

I should have walked through the commits. 😕 
> I have just opted to update the other two screenshots to make the all subsection look the same. But, I can revert if you prefer.

No. It's done! let's not waste our resources on this. And I suspected that was the reason.
OK to merge!

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 2]
```diff
@@ -728,14 +728,14 @@ GDAL Settings
 
 `GDAL <https://gdal.org>`_ is a data exchange library for vector and raster
 files. It provides drivers to read and or write data in different formats.
```
Comment (havatv) [inline docs/user_manual/introduction/qgis_configuration.rst:730]:
```suggestion
`GDAL <https://gdal.org>`_ is a data exchange library sfor geospatial data that supports a
large number of vector and raster formats.
It provides drivers to read and (often) write data in these formats.
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 3]
```diff
@@ -728,14 +728,14 @@ GDAL Settings
...
-The :guilabel:`GDAL` tab currently exposes the drivers for raster formats with
-their capabilities.
+The :guilabel:`GDAL` tab currently exposes the drivers for raster and vector
```
Comment (havatv) [inline docs/user_manual/introduction/qgis_configuration.rst:731]:
```suggestion
The :guilabel:`GDAL` tab exposes the drivers for raster and vector
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 4]
```diff
@@ -777,22 +777,22 @@ and write access:
...
 
-GDAL drivers
-............
+GDAL raster and vector drivers
+..............................
 
-In this frame, you can define which GDAL driver is to use to read and/or
+The :guilabel:`Raster Drivers` and :guilabel:`Vector Drivers` (in a separated
+tab) allow you to define which GDAL driver is enabled to read and/or
 write files, as in some cases more than one GDAL driver is available.
```
Comment (havatv) [inline docs/user_manual/introduction/qgis_configuration.rst:785]:
```suggestion
The :guilabel:`Raster Drivers` and :guilabel:`Vector Drivers` (in separate tabs)
allow you to configure the GDAL drivers.
```
Not sure what can and can not do here,but I can't find a  way to enable / disable.

File: docs/user_manual/managing_data_source/opening_data.rst [hunk 5]
```diff
@@ -48,9 +48,8 @@ More than 80 vector and 140 raster formats are supported by
    installation of your OS may not have been built to support the format you
    want to use. To see the list of available formats, run the command line
    ``ogrinfo --formats`` (for vector) and ``gdalinfo --formats`` (for raster),
-   or check :menuselection:`Settings --> Options --> GDAL` menu (for raster)
-   in QGIS.
-   
+   or check :menuselection:`Settings --> Options --> GDAL` menu in QGIS.
```
Comment (havatv) [inline docs/user_manual/managing_data_source/opening_data.rst:51]:
```suggestion
   or check the :menuselection:`Settings --> Options --> GDAL` menu in QGIS.
```

File: docs/user_manual/managing_data_source/opening_data.rst [hunk 6]
```diff
@@ -232,7 +231,7 @@ the layers have been added to the project):
 To add a layer to the project using the :guilabel:`Browser`:
 
 #. Enable the :guilabel:`Browser` as described above.
-   A browser tree with your file system, databases and web services is 
+   A browser tree with your file system, databases and web services is
    displayed.
    You may need to connect databases and web services before they appear
    (see dedicated sections).
```
Comment (havatv) [inline docs/user_manual/managing_data_source/opening_data.rst:237]: Should we provide links to these sections?

**Review by havatv — APPROVED** [review]
Looks good to me.
I have some suggestions /questions / comments  I hope you will consider

File: docs/user_manual/introduction/qgis_configuration.rst (hunk 4 above)
Comment (SrNetoChan) [reply docs/user_manual/introduction/qgis_configuration.rst:785]: Before each driver, there's a checkbox. You can disable each of the drivers.

File: docs/user_manual/managing_data_source/opening_data.rst (hunk 6 above)
Comment (SrNetoChan) [reply docs/user_manual/managing_data_source/opening_data.rst:237]: This change is just my text editor cleaning up. Has no relation with the original objective.

File: docs/user_manual/introduction/qgis_configuration.rst (hunk 4 above)
Comment (havatv) [reply docs/user_manual/introduction/qgis_configuration.rst:786]: > Before each driver, there's a checkbox. You can disable each of the drivers.

Thanks ( for some reason, I did not notice the checkboxes).
I have now updated the suggestion.

Comment (SrNetoChan) [reply docs/user_manual/introduction/qgis_configuration.rst:786]: @havatv I am not so sure about your suggestion. It removes the explanation of why someone would want to disable a driver. Besides, you say you can configure the GDAL drivers, but there isn't much you can configure. Unless you talking about the double-click trick for rasters.

Comment (havatv) [reply docs/user_manual/introduction/qgis_configuration.rst:786]: > @havatv I am not so sure about your suggestion. It removes the explanation of why someone would want to disable a driver. Besides, you say you can configure the GDAL drivers, but there isn't much you can configure. Unless you talking about the double-click trick for rasters.

OK. But what about *Edit Create Options* / *Edit Pyramids  Options* (3.14)

Comment (SrNetoChan) [reply docs/user_manual/introduction/qgis_configuration.rst:786]: > > @havatv I am not so sure about your suggestion. It removes the explanation of why someone would want to disable a driver. Besides, you say you can configure the GDAL drivers, but there isn't much you can configure. Unless you talking about the double-click trick for rasters.
> 
> OK. But what about _Edit Create Options_ / _Edit Pyramids Options_ (3.14)

That is described in the previous section. This text is about the Raster drivers and Vector Drivers lists.

**Comment by DelazJ** [comment]
Seems everything is solved here...

**Pull request merged**